vassal-app/src/main/java/VASSAL/tools/DebugControls.java

```diff
@@ -85,12 +90,33 @@ public Point getCursorLocation() {
     return cursorLocation;
   }
 
-  public void setCursorLocation(Point pt) {
+  public void setCursorLocation(Point pt, Map map) {
     cursorLocation = pt;
+
+    if (map != null) {
+      final Board b = map.findBoard(pt);
+      if (b != null) {
+        final Rectangle r = b.bounds();
+        final Point bp = new Point(pt);
+        bp.translate(-r.x, -r.y);
+        cursorLocationBoard = bp;
+      }
+      else {
+        cursorLocationBoard = pt;
+      }
+    }
+    else {
+      cursorLocationBoard = pt;
+    }
```

Comment:
Most of this is repeated below... I wonder if there's a way to extract both this and the nearly identical block lower down to a function?

Reply:
Fixed